src/components/dashboard/DashboardRouter/DashboardRouter.js

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import { useSelector } from 'react-redux';
+import { Switch, Route, Redirect, useLocation } from 'react-router-dom';
+
+import { Box, Center, ChakraProvider, Grid, Spinner } from '@chakra-ui/react';
+
+import dashboardTheme from 'themes/dashboard';
+
+import Dashboard from 'pages/Dashboard/Dashboard';
+import Invite from 'pages/Invite/Invite';
+import LoginRegister from 'pages/LoginRegister/LoginRegister';
+import TeamView from 'pages/TeamView/TeamView';
+
+import Navbar, { NAVBAR_WIDTH } from '../Navbar/Navbar';
+import ProtectedRoute from '../ProtectedRoute/ProtectedRoute';
+
+const DashboardRouter = () => {
+  const { authenticating, user } = useSelector(state => state.auth);
+  const location = useLocation();
+
+  return (
+    <ChakraProvider theme={dashboardTheme}>
+      {/* Show spinner while authenticating */}
+      {authenticating ? (
+        <Center h="100vh">
+          <Spinner size="xl" />
+        </Center>
+      ) : (
+        <Switch>
+          {/* ProtectedRoute will redirect to /login if not signed in
+          It also sets location.state.from
+          After signing in, redirect to the original path 
+          */}
+          <Route exact path="/login">
+            {user ? <Redirect to={location.state?.from?.pathname ?? '/'} /> : <LoginRegister />}
```

Comment:
I refactored the router into its own file, and used `location.state` to redirect to a page after logging in. This works for all pages, not only the invite page!